### PR TITLE
[android][local-authentication] Add the ability to set confirmation requirement

### PR DESCRIPTION
**Pull request by fguitton**
Bonjour !
Good day to everyone
### 🌴 

# Why

In the current setup iOS and Android behave differently when using FaceId / IrisScan.
Namely by default iOS will transparently validate a face and get back to the app. Android will by default require a confirmation by button press to the user.

The propose change does not explicitly mean to align the behavior at the level of Expo, but rather offer developers the choice while respecting the Android API default.

# How

By adding a new optional variable to the API which calls the relevant [`setConfirmationRequired`](https://developer.android.com/reference/kotlin/androidx/biometric/BiometricPrompt.PromptInfo.Builder#setConfirmationRequired\(kotlin.Boolean\)) function of `PromptInfo.Builder`

# Test Plan

In an Expo application call the function:
```typescript
await LocalAuthentication.authenticateAsync({
    requiredConfirmation: false
});
```
Verify the prompt does not require button press upon face scan.
_Note that this just passes a hint to the system, which the system may then ignore. For example, a value of false may be ignored if the user has disabled implicit authentication in Settings, or if it does not apply to a particular modality (e.g. fingerprint)._

# Checklist

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [X] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [X] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-local-authentication/android/src/main/java/expo/modules/localauthentication/LocalAuthenticationModule.kt [hunk 1]
```diff
@@ -206,6 +206,11 @@ class LocalAuthenticationModule(context: Context) : ExportedModule(context), Act
         } else {
           false
         }
+        val requireConfirmation = if (options.containsKey("requireConfirmation")) {
+          options["requireConfirmation"] as Boolean?
+        } else {
+          true
+        }
```
Comment (lukmccall) [inline packages/expo-local-authentication/android/src/main/java/expo/modules/localauthentication/LocalAuthenticationModule.kt:213]:
```suggestion
        val requireConfirmation = options["requireConfirmation"] as? Boolean ?: true
```

File: packages/expo-local-authentication/android/src/main/java/expo/modules/localauthentication/LocalAuthenticationModule.kt [hunk 2]
```diff
@@ -224,6 +229,8 @@ class LocalAuthenticationModule(context: Context) : ExportedModule(context), Act
               or BiometricManager.Authenticators.DEVICE_CREDENTIAL
           )
         }
+        if (requireConfirmation == false)
+          promptInfoBuilder.setConfirmationRequired(false)
```
Comment (lukmccall) [inline packages/expo-local-authentication/android/src/main/java/expo/modules/localauthentication/LocalAuthenticationModule.kt:233]:
```suggestion
        promptInfoBuilder.setConfirmationRequired(requireConfirmation)
```

File: packages/expo-local-authentication/CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,8 @@
 
 ### 🎉 New features
 
+- Add the ability to set confirmation requirement on Android ([#16793](https://github.com/expo/expo/pull/16793) by [@fguitton](https://github.com/fguitton))
```
Comment (lukmccall) [inline packages/expo-local-authentication/CHANGELOG.md:9]:
```suggestion
- Add the ability to set confirmation requirement on Android. ([#16793](https://github.com/expo/expo/pull/16793) by [@fguitton](https://github.com/fguitton))
```

**Review by lukmccall — CHANGES_REQUESTED** [review]
Thanks for your contribution 🏅
I've left couple of comments ;)

**Comment by fguitton** [comment]
> I've left couple of comments

It makes a lot of sense, I've now brought the changes and pushed.

**Review by lukmccall — APPROVED** [review]

**Pull request merged**